# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/gamepad/ipc/GamepadEventChannelChild.h
# Commit: 8132dd939023
# Full Hash: 8132dd9390233328c8da7072f36859424a2b079d
# Author: Chris Martin <cmartin@mozilla.com>
# Date: 2020-10-14 21:42:48
# Regressor Bug: 1657404
# File Overlap Count: 2
# Description:
#   Bug 1657404 - Change PGamepadEventChannel to "refcounted protocol" r=handyman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D93119
# ==============================================================================

diff -r 186ca9e93e1d -r 8132dd939023 dom/gamepad/ipc/GamepadEventChannelChild.h
--- a/dom/gamepad/ipc/GamepadEventChannelChild.h	Wed Oct 07 11:34:02 2020 +0000
+++ b/dom/gamepad/ipc/GamepadEventChannelChild.h	Wed Oct 14 15:58:36 2020 +0000
@@ -14,15 +14,24 @@
 
 class GamepadEventChannelChild final : public PGamepadEventChannelChild {
  public:
-  GamepadEventChannelChild() = default;
-  ~GamepadEventChannelChild() = default;
+  NS_INLINE_DECL_THREADSAFE_REFCOUNTING(GamepadEventChannelChild, override)
+
+  static already_AddRefed<GamepadEventChannelChild> Create();
 
   mozilla::ipc::IPCResult RecvGamepadUpdate(
       const GamepadChangeEvent& aGamepadEvent);
   mozilla::ipc::IPCResult RecvReplyGamepadPromise(const uint32_t& aPromiseID);
   void AddPromise(const uint32_t& aID, dom::Promise* aPromise);
 
+  GamepadEventChannelChild(const GamepadEventChannelChild&) = delete;
+  GamepadEventChannelChild(GamepadEventChannelChild&&) = delete;
+  GamepadEventChannelChild& operator=(const GamepadEventChannelChild&) = delete;
+  GamepadEventChannelChild& operator=(GamepadEventChannelChild&&) = delete;
+
  private:
+  GamepadEventChannelChild() = default;
+  ~GamepadEventChannelChild() = default;
+
   nsRefPtrHashtable<nsUint32HashKey, dom::Promise> mPromiseList;
 };
 